source/constructing-lattices.md

```diff
@@ -0,0 +1,306 @@
+(c:construction)=
+# Constructing Lattices
+
+%---------------------------------------------------------------------------------------------------
+(s:beamlines)=
+## BeamLines
+
+A lattice [`branch`](#s:branches) is constructed from a `BeamLine`. A BeamLine is essentially 
+an ordered array of elements. Each element is either a lattice element or a `BeamLine`. 
+A BeamLine that is contained within another BeamLine is called a `subline`
+of the containing BeamLine.
+The top level `BeamLine` from which a branch is constructed is called a `root Beamline`. 
+
+The components of a BeamLine are:
+```{code} yaml
+name        # String: Name of the BeamLine. Optional.
+multipass   # Bool: Multipass line or not? Optional. Default is False.
+length      # [m]: Length of the BeamLine. Optional.
+line        # Ordered list: List of elements. Required.
+reference_element  # Reference element. Optional.
+```
+
+The `name` component is a string that can be used to reference the `BeamLine`.
+
+The optional `mutipass` component is a boolean describing whether the `BeamLine` is part of 
+a [`multipass`](#c:multipass) construct. The default is False.
+
+The optional `length` component gives the length of the `BeamLine`. 
+If `length` is not given, the BeamLine ends at the downstream end of the final
+element in the `line` and with this the length of the BeamLine can be calculated.
+
+The optional `referece_element` component is used to position sublines.
+See below.
+This referece element must have zero length.
+
+The `line` component of a BeamLine holds an ordered list of `items`. 
+Each `item` represents one (or more if there is a `repeat` count) lattice element or
+BeamLine. 
+
+A line `item` can have a value which is the name of a lattice element or `BeamLine`, 
```

Comment:
Minor stylistic suggestion: replace "which" with "that".

Reply:
Done.